### PR TITLE
Apply suggested style changes to C# code

### PR DESCRIPTION
**Pull request by wilyle**
This PR applies suggested style changes to C# code from the official .NET tooling. This also adds a step to the .NET CI workflow to perform these checks

Also makes a few documentation changes related to one of the CA2254 changes since there's less information in the exception now. That info is still being logged though and now the exception is documented better, so I don't think we need that info in the exception itself

### CODE REVIEW

**Review by ashbeitz — DISMISSED** [review]
Approved.

File: cloud_connectors/azure/mqtt_connector/azure_function/src/run.cs [hunk 1]
```diff
@@ -57,11 +69,14 @@ public MQTTConnectorAzureFunction(ILogger<MQTTConnectorAzureFunction> logger)
         /// </summary>
         /// <param name="dataTypeName">the name of the data type.
         /// <returns>Returns a task for updating a digital twin instance.</returns>
```
Comment (jorchiu) [inline cloud_connectors/azure/mqtt_connector/azure_function/src/run.cs:71]: Maybe add a comment for `NotSupportedException`.

```suggestion
        /// <exception cref="NotSupportedException">Thrown if the data type is not supported.</exception>
       /// <returns>Returns a task for updating a digital twin instance.</returns>
```

**Review by devkelley — APPROVED** [review]

File: cloud_connectors/azure/mqtt_connector/azure_function/src/run.cs [hunk 2]
```diff
@@ -19,20 +20,30 @@ namespace Microsoft.ESDV.CloudConnector.Azure {
...
+
+        [JsonPropertyName("instance_property_path")]
+        public string InstancePropertyPath { get; set; }
+
+        [JsonPropertyName("data")]
+        public string Data { get; set; }
     }
 
-    public class MQTTConnectorAzureFunction {
+    public class MQTTConnectorAzureFunction
```
Comment (ashbeitz) [inline cloud_connectors/azure/mqtt_connector/azure_function/src/run.cs:38]: Please add a class method.

File: cloud_connectors/azure/mqtt_connector/azure_function/tests/MQTTConnectorAzureFunction.Tests.cs [hunk 3]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the MIT license.
 // SPDX-License-Identifier: MIT
```
Comment (ashbeitz) [inline cloud_connectors/azure/mqtt_connector/azure_function/tests/MQTTConnectorAzureFunction.Tests.cs:3]: What about the Microsoft part of the license?  It seems to be missing.

**Review by ashbeitz — COMMENTED** [review]
A few minor comments.

File: cloud_connectors/azure/mqtt_connector/azure_function/src/run.cs (hunk 2 above)
Comment (wilyle) [reply cloud_connectors/azure/mqtt_connector/azure_function/src/run.cs:38]: I'm not sure what you're asking here

File: cloud_connectors/azure/mqtt_connector/azure_function/tests/MQTTConnectorAzureFunction.Tests.cs (hunk 3 above)
Comment (wilyle) [reply cloud_connectors/azure/mqtt_connector/azure_function/tests/MQTTConnectorAzureFunction.Tests.cs:3]: it's already there on line 1

File: cloud_connectors/azure/mqtt_connector/azure_function/src/run.cs (hunk 2 above)
Comment (ashbeitz) [reply cloud_connectors/azure/mqtt_connector/azure_function/src/run.cs:38]: Sorry, meant "Please add a class comment".

**Pull request merged**